### ISSUE TITLE
Losing TR in matchmaking?

### ISSUE THREAD

**Issue by AresTargaryen** [tetrio/issues#978]
### Checklist

- [X] I've **checked both [open and recently closed issues](https://github.com/tetrio/issues/issues?q=is%3Aissue)** to ensure this issue is not already reported.
- [X] I am using the **newest version** of TETR.IO.
- [X] I am using a **supported browser** or the Desktop Client.
- [X] I am not using anything that **modifies game code** (e.g. TETR.IO PLUS).
- [X] I am certain this is a **bug**, not a feature request.

### OS and Version

Windows 10

### Browser and version

Tetrio Desktop: version 6.2.1

### Describe the bug/issue

So basically, I entered Tetra League matchmaking, and literally after that I got a connection error **(while still in matchmaking**). After that i checked my TR and it went down from ~6600 to 6415 and currently I am on demotion on next loss.
I've checked the Tetra Channel but on the record it didn't count anything, my win rate didn't drop, I just lost TR.

### Expected behavior

Find, start and finish a match succsesfully with TR gain if won and TR drop if lost.

**Comment by ZaptorZap** [comment]
Sorry, but this is intentional—you **did** match against someone, you just couldn't tell on your own screen as you had disconnected before you could receive any match information, and the server could not catch up to your disconnection in time. This's called "canceling" a game—and you're penalized as it's _technically_ possible to "dodge" matchups during the period inbetween matchmaking and matchups. Your opponent saw a special disconnection message informing them that _their TR_ was not adjusted. (for comparison, Puyo Puyo Tetris and Tetris Effect: Connected let you do the same far easier without penalization. (Don't like the username that comes up as your enemy? Press the `B` button a couple times: they'll go away!) As far as I can tell, nobody's very invested in either of those rating systems, although that's mostly coincidence since the discrepancy is driven by other factors)
